profiles/common/modules/features/nexteuropa_editorial/includes/nexteuropa_editorial.tracking.inc

```diff
@@ -0,0 +1,429 @@
+      return t('The form cannot be saved because of tracked changes existing in the @lang versions.',
+        $message_param);
+  }
+}
+
+/**
+ * Gets the widget type used by a specific field instance.
+ *
+ * @param array $field_instance
+ *   The field instance for which the related widget type.
+ *
+ * @return bool|string
+ *   Widget type; otherwise FALSE.
+ */
+function _nexteuropa_editorial_field_get_widget_type($field_instance) {
+  $widget_info = _nexteuropa_editorial_field_get_widget_info($field_instance);
+
+  if ($widget_info) {
+    return $widget_info['type'];
+  }
+
+  return FALSE;
+}
+
+/**
+ * Gets widget information for a specific field instance.
+ *
+ * @param array $field_instance
+ *   The field instance for which we need information.
+ *
+ * @return bool|array
+ *   Widget information; otherwise FALSE.
+ */
+function _nexteuropa_editorial_field_get_widget_info($field_instance) {
+  if (isset($field_instance['widget'])) {
+    return $field_instance['widget'];
+  }
+
+  return FALSE;
+}
+
+/**
+ * Checks if the saving of the entity must be blocked or not.
+ *
+ * @param object $entity
+ *   The entity to check.
+ * @param bool|FALSE $current_wbm_state
+ *   The entity current workbench moderation state (node only).
+ *
+ * @return bool
+ *   TRUE if the saving must be blocked; otherwise FALSE.
+ */
+function _nexteuropa_editorial_tracking_saving_to_block($entity, $current_wbm_state = FALSE) {
+  // So far, only nodes are support.
+  if ($entity->entity_type == 'node') {
+    $settings = _nexteuropa_editorial_wysiwyg_tracking_settings($entity);
+
+    if ($settings['is_activate']) {
+      if (empty($settings['wbm_states'])) {
+        return $settings['saving_blocking'];
+      }
+      elseif (in_array($current_wbm_state, $settings['wbm_states'])) {
+        return TRUE;
+      }
+    }
+  }
+
+  return FALSE;
+}
+
+/**
+ * Get the WYSIWYG tracking workflow settings for the entity.
+ *
+ * @param object $entity
+ *   The entity from which retrieving settings.
+ *
+ * @return array
+ *   An associative array containing:
+ *   - is_activate: flag indicating if controls are active about change
+ *       tracking.
+ *   - saving_blocking: flag indicating if savings must blocked if tracked
+ *      changes exist.
+ *   - wbm_states: (only node) it gives the list of state to take into account.
+ */
+function _nexteuropa_editorial_wysiwyg_tracking_settings($entity = NULL) {
+  if ($entity && isset($entity->entity_type)) {
+    $type = $entity->entity_type;
+    $cached_settings = &drupal_static(__FUNCTION__ . '_' . $type);
+
+    // So far, only nodes are supported.
+    if (empty($cached_settings) && ($type == 'node')) {
+      $status = variable_get('nexteuropa_editorial_tracking_status');
+      $states = variable_get('nexteuropa_editorial_tracking_wbm_states', array());
+      $states_to_check = array();
+      foreach ($states as $state_name => $state_status) {
```

Comment:
Do we need to perform this loop. It is just removing key and creates new array just with values.
Maybe http://php.net/manual/en/function.array-values.php could be an alternative for that.